config/data_scrubber.go

```diff
@@ -90,6 +104,40 @@ func compileStringsToRegex(words []string) []*regexp.Regexp {
 	return compiledRegexps
 }
 
+// createProcessKey returns an unique identifier for a given process
+func createProcessKey(p *process.FilledProcess) string {
+	var b bytes.Buffer
+	b.WriteString("p:")
+	b.WriteString(strconv.Itoa(int(p.Pid)))
+	b.WriteString("|c:")
+	b.WriteString(strconv.Itoa(int(p.CreateTime)))
+
+	return b.String()
+}
+
+// ScrubCmdlineWithCache uses a cache memory to avoid scrubbing already known
+// process' cmdlines
+func (ds *DataScrubber) ScrubCmdlineWithCache(p *process.FilledProcess) []string {
```

Comment:
Call this function `ScrubProcessCommand` (or something like that) -- theres no need to expose that it is cached. You can make the other one unexported.